docs/versionSelector.js

```diff
@@ -0,0 +1,69 @@
+import {
+  LitElement,
+  css,
+  html,
+} from "https://cdn.jsdelivr.net/gh/lit/dist@3/all/lit-all.min.js";
+
+class VersionSelector extends LitElement {
+  static styles = css`
+    select {
+      padding: 5px;
+      margin: 10px 0;
+      font-size: 100%;
+    }
+  `;
+
+  static properties = {
+    versions: { type: Array },
+  };
+
+  constructor() {
+    super();
+    this.versions = [];
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.fetchVersions();
+  }
+
+  async fetchVersions() {
+    try {
+      const response = await fetch(
+        "https://martinvonz.github.io/jj/versions.json",
+      );
+      if (response.ok) {
+        this.versions = await response.json();
+      }
+    } catch (error) {
+      console.error("Error fetching versions:", error);
+    }
+  }
+
+  handleVersionChange(e) {
+    const selectedVersion = e.target.value;
+    const currentUrl = new URL(window.location);
+    const newUrl = currentUrl.href
+      .replace(
+        // /(\/jj\/)[^\/]+/,
+        /http:\/\/localhost:3000(.*)/,
+        `https://martinvonz.github.io/jj/${selectedVersion}$1`,
```

Comment:
Just for testing; this will be More Proper in the prod version

(I should look at mkdocs's implementation perhaps)

Reply:
If you go to `prerelease/bookmarks/`, and switch to a pre-bookmarks `jj` version, will this show an ugly 404?

A lot of the complexity in the mkdocs-material version switcher functionality comes from trying to avoid this. See e.g. https://github.com/squidfunk/mkdocs-material/pull/7338 (which has tests and thus might be slightly easier to follow than what ended up merged).

It is possible that fixing it is not worth the complexity, but then we'd need to make our 404 page much nicer.